lib/samples/edit_feature_attachments/edit_feature_attachments_sample.dart

```diff
@@ -0,0 +1,384 @@
+  }
+
+  // view an attachment from the selected feature in a dialog.
+  void viewAttachment(Attachment attachment) async {
+    setState(() => isLoading = true);
+
+    final data = await attachment.fetchData();
+    setState(() => isLoading = false);
+
+    // display the attachment image/pdf file in a dialog.
+    setState(() {
+      showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(
+            title: Text(attachment.name),
+            content: SizedBox(
+              height: 300,
+              child: Image.memory(data),
+            ),
+            actions: [
+              IconButton(
+                icon: const Icon(Icons.save),
+                tooltip: 'Save',
+                onPressed: () {
+                  // save the attachment to the device.
+                  FilePicker.platform.saveFile(
+                    dialogTitle: 'Save Attachment',
+                    fileName: attachment.name,
+                    bytes: data,
+                    lockParentWindow: true,
+                  );
+                },
+              ),
+              IconButton(
+                icon: const Icon(Icons.close),
+                tooltip: 'Close',
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      );
+    });
+  }
+
+  // add an attachment to the selected feature by FilePicker.
+  void addAttachment() async {
+    var status = false;
+    final result = await FilePicker.platform.pickFiles(
+      allowMultiple: false,
+      type: FileType.custom,
+      onFileLoading: (filePickerStatus) {
+        if (filePickerStatus == FilePickerStatus.done) {
+          status = true;
+        }
+      },
+      allowedExtensions: ['png', 'jpg', 'jpeg', 'pdf', 'txt'],
+    );
+
+    if (result != null && status) {
+      setState(() => isLoading = true);
+
+      final platformFile = result.files.single;
+      final file = File(platformFile.path!);
+
+      // Get the context type for the file
+      final fileExtension = platformFile.extension?.toLowerCase();
+      final contentType = getContextType(fileExtension ?? 'default');
+      final fileBytes = await file.readAsBytes();
+
+      await widget.arcGISFeature.addAttachment(
+        name: platformFile.name,
+        contentType: contentType,
+        data: fileBytes,
+      );
+      await widget.applyEdits(widget.arcGISFeature);
+      _loadAttachments();
+
+      setState(() => isLoading = false);
+    }
+  }
+
+  // load and update the attachments for the selected feature
+  Future<void> _loadAttachments() async {
+    try {
+      var fetchedAttachments = await widget.arcGISFeature.fetchAttachments();
+      setState(() {
+        attachments = fetchedAttachments;
+        isLoading = false;
+      });
+    } catch (e) {
+      setState(() {
+        _showErrorDialog(context, e);
```

Comment:
I think the `_showErrorDialog` should be moved outside of the call to `setState`, probably after it.
```dart
setState(() => isLoading = false);
_showErrorDialog(context, e);
```